cartoframes/assets/vector.html

```diff
@@ -0,0 +1,204 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <title>CARTO VL + CARTOframes</title>
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <meta charset="UTF-8">
+  <!-- Include CARTO VL JS -->
+  <script src="@@CARTO_VL_PATH@@"></script>
+  <!-- Include Mapbox GL JS -->
+  <script src="https://api.tiles.mapbox.com/mapbox-gl-js/v0.52.0/mapbox-gl.js"></script>
+  <!-- Include Mapbox GL CSS -->
+  <link href="https://api.tiles.mapbox.com/mapbox-gl-js/v0.52.0/mapbox-gl.css" rel="stylesheet" />
+
+  <!-- Include Airship -->
+  <script src="@@AIRSHIP_COMPONENTS_PATH@@"></script>
+  <script src="@@AIRSHIP_BRIDGE_PATH@@"></script>
+  <link href="@@AIRSHIP_STYLES_PATH@@" rel="stylesheet">
+  <link href="@@AIRSHIP_ICONS_PATH@@" rel="stylesheet">
+
+  <style>
+    body {
+      margin: 0;
+      padding: 0;
+    }
+
+    #map {
+      position: absolute;
+      height: 100%;
+      width: 100%;
+    }
+  </style>
+</head>
+<body class="as-app-body as-app">
+  <div class="as-content">
+    <main class="as-main">
+      <div class="as-map-area">
+        <div id="map"></div>
+        <div class="as-map-panels">
+          <div class="as-panel as-panel--right as-panel--top">
+            <div class="as-panel__element" id="legends" style="display: none;">
+            </div>
+          </div> <!-- as-panel -->
+        </div> <!-- as-map-panels -->
+      </div> <!-- as-map-area -->
+    </main> <!-- as-main -->
+  </div> <!-- as-content -->
+
+  <script>
+    const BASEMAPS = {
+        DarkMatter: carto.basemaps.darkmatter,
+        Voyager: carto.basemaps.voyager,
+        Positron: carto.basemaps.positron
+    };
+    if ("@@MAPBOXTOKEN@@") {
+      mapboxgl.accessToken = "@@MAPBOXTOKEN@@";
+    }
+    // Fetch CARTO basemap if it's there, else try to use other supplied style
+    const map = new mapboxgl.Map({
+      container: 'map',
+      style: BASEMAPS['@@BASEMAPSTYLE@@'] || "@@BASEMAPSTYLE@@",
+      zoom: 9,
+      dragRotate: false
+    });
+
+    let credentials = @@CREDENTIALS@@;
+    carto.setDefaultAuth({
+      username: credentials['username'],
+      apiKey: credentials['api_key'] || 'default_public'
+    });
+    carto.setDefaultConfig({
+        serverURL: credentials['base_url'] || `https://${credentials['user']}.carto.com/`
+    });
+    var sources = @@SOURCES@@;
```

Comment:
Wasn't this replaced by Jinja?

Reply:
Yes, but we left the default _vector, contrib_ code in order to remove it in a separate task. Here we have the issue: https://github.com/CartoDB/cartoframes/issues/597